addons/website/static/src/components/website_loader/website_loader.js

```diff
@@ -1,29 +1,214 @@
+        // Update the state only if it is moving forward (as the calls are
+        // async but not awaited within the interval, we can't be sure they
+        // return sequentially)
+        if (!this.featuresInstallInfo.total
+            || this.featuresInstallInfo.nbInstalled < installInfo.nbInstalled) {
+            Object.assign(
+                this.featuresInstallInfo,
+                { nbInstalled: installInfo.nbInstalled, total: installInfo.total }
+            );
+        }
+    };
+
+    getWaitingTips = (selectedFeatures) => {
+        const addFeatureTip = (id, name, title, description) => {
+            const tipNameHighlighted = sprintf(
+                "<span class='o_website_loader_text-highlight'>%s</span>", name
+            );
+
+            return {
+                condition: selectedFeatures.includes(id),
+                title: markup(sprintf(title, tipNameHighlighted)),
+                description: description,
+            };
+        }
+
+        const tipsList = [
+            {
+                condition: true,
+                title: _t("Building your website."),
+                description: _t("Stay tuned for an exciting new online presence."),
+            },
+            addFeatureTip(
+                5,
+                _t("blog"),
+                _t("Enabling your %s."),
+                _t("Share your thoughts and ideas with the world.")
+            ),
+            addFeatureTip(
+                7,
+                _t("recruitment platform"),
+                _t("Integrating your %s."),
+                _t("Find the best talent for your team.")
+            ),
+            addFeatureTip(
+                8,
+                _t("online store"),
+                _t("Activating your %s."),
+                _t("Start selling your products and services today.")
+            ),
+            addFeatureTip(
+                9,
+                _t("online appointment system"),
+                _t("Configuring your %s."),
+                _t("Make it easy for clients to book appointments with you.")
+            ),
+            addFeatureTip(
+                10,
+                _t("forum"),
+                _t("Setting up your %s."),
+                _t("Engage with your community and build relationships.")
+            ),
+            addFeatureTip(
+                12,
+                _t("e-learning platform"),
+                _t("Installing your %s."),
+                _t("Offer online courses and learning opportunities.")
+            ),
+            {
+                condition: !!selectedFeatures.length,
+                title: _t("Activating the last features."),
+                description: _t("A bit more patience as your website takes shape."),
+            },
+        ];
+        return (
+            tipsList.filter((tip, i) => {
+                tip.id = i;
+                tip.show = i === 0;
+                return tip.condition;
+            })
+        );
+    };
```

Comment:
- `selectedFeatures` is only supposed to change once, but the method was indeed messy at best. I'll extract the tips and the refactored function outside of it. But you'll see I opted for a plain array for 2 reasons: 1) it's easier to manipulate and 2) to avoid making useless manipulations later on, the 2 non-feature related tips are also in there.
- I thought about using `t-out` to inject the name inside the title, etc. directly from the XML, but I would need to be able to replace `%s` from the XML to do so and I don't think I can? (I need the `%s` for translations, where the word order might differ.)

Reply:
- As said in my "behavior" comment, it was called at least twice (as we discussed it :wink:)
- You are right, I didn't think about this translation issue, it can stay like that then :+1: